Frontend/spm-proj/src/components/SearchBar.vue

```diff
@@ -0,0 +1,202 @@
+                                :value="skill.name"
+                                v-model="selectedSkills"
+                                style="max-width: 150px;"
+                            />
+                            <label class="form-check-label" :for="'skill_' + skill.id">{{ skill.name }}</label>
+                            </div>
+                        </div>
+                        </div>
+                        <a class="btn btn-link mt-5" @click="clearSelection">Clear Selection</a>
+                    </div>
+                    </form>
+                </div>
+                </div>
+                <button type="button" class="btn btn-outline-secondary" @click="triggerSearch" id="searchButton">Search
+                <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-search" viewBox="0 0 16 16">
+                    <path d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z" />
+                </svg>
+                </button>
+            </div>
+            </div>
+        </div>
+        </div>
+    </div>
+    </div>
+</template>
+
+<script>
+import 'bootstrap/dist/css/bootstrap.css'; // Import Bootstrap 4 CSS
+import 'jquery/dist/jquery.min.js'; // Import jQuery
+import 'bootstrap/dist/js/bootstrap.min.js'; // Import Bootstrap 4 JS
+
+export default {
+    name: 'SearchBar',
+    data() {
+    return {
+        isSkillsDropdownOpen: false,
+        isDepartmentsDropdownOpen: false,
+        skills: [
+        { id: 1, name: 'JavaScript' },
+        { id: 2, name: 'HTML' },
+        { id: 3, name: 'CSS' },
+        { id: 4, name: 'Analytics' },
+        { id: 5, name: 'Excel' },
+        { id: 6, name: 'UX/UI' },
+        // Add more skills as needed
+        ],
+        selectedSkills: [],
+        departments: [
+        { id: 1, name: 'HR' },
+        { id: 2, name: 'Marketing' },
+        { id: 3, name: 'Finance' },
+        { id: 4, name: 'IT' },
+        // Add more departments as needed
+        ],
+        selectedDepartments: [],
+        searchKeyword: '', 
+    };
+    },
+    methods: {
+    toggleSkillsDropdown() {
+        this.isSkillsDropdownOpen = !this.isSkillsDropdownOpen;
+    },
+    toggleDepartmentsDropdown() {
+        this.isDepartmentsDropdownOpen = !this.isDepartmentsDropdownOpen;
+    },
+    clearSelection() {
+        this.selectedSkills = [];
+        this.selectedDepartments = [];
+    },
+    triggerSearch() {
+        this.$emit('search-request', {
+            keyword: this.searchKeyword,
+            selectedDepartments: this.selectedDepartments,
+            selectedSkills: this.selectedSkills,
+        });
+    },
+    },
+    mounted(){
+        const searchButton = document.getElementById('searchButton');
+        searchButton.addEventListener('click', this.triggerSearch);
```

Comment:
for addEventListener method here, it throws a typeerror in the screenshot below
<img width="322" alt="Screenshot 2023-09-30 173842" src="https://github.com/L1uM4rt1n/SPM-Project/assets/96768015/6ac42f25-57fa-4278-9105-4b44b72dbafa">

make sure that the element being targeted by the method exists in the DOM when the components is mounted, if not wrap the method in a setTimeout function with a delay of 0 to help ensure the element is fully rendered before the method is called